### PR TITLE
Sts v2

### PR DESCRIPTION
**Pull request by cviecco**
Moving cloudfate to use sts v2

### CODE REVIEW

**Review by kyleip — APPROVED** [review]
LGTM

File: broker/aws/impl.go [hunk 1]
```diff
@@ -128,21 +128,21 @@ func (b *Broker) accountHumanNameFromName(accountName string) (string, error) {
...
 		b.logger.Printf("Unable to get master credentials: %s\n", err)
 		return nil
 	}
-	masterSession, err := session.NewSession(
-		aws.NewConfig().WithCredentials(sessionCredentials).WithRegion(region))
-	if err != nil {
-		return err
+	if region == "" {
+		b.logger.Printf("Unable to get master credentials empty region")
+		return nil
```
Comment (rgooch) [inline broker/aws/impl.go:139]: Does STS definitely require a Region to be specified?

File: broker/aws/impl.go [hunk 2]
```diff
@@ -235,75 +235,74 @@ func (b *Broker) loadCredentialsFrombytes(credentials []byte) error {
...
 		if profileName == masterAWSProfileName {
-			return b.getCredentialsFromMetaData()
+			return b.getCredentialsProviderFromMetaData()
 		}
 		return nil, "", fmt.Errorf("invalid profileName: %s", profileName)
 	}
-	sessionCredentials := credentials.NewStaticCredentials(
+	provider := credentials.NewStaticCredentialsProvider(
 		profileEntry.AccessKeyID, profileEntry.SecretAccessKey, "")
-	b.logger.Printf("Created credentials object for static profile: %s\n",
```
Comment (rgooch) [inline broker/aws/impl.go:249]: Why remove this logging? It's useful.

File: broker/aws/impl.go [hunk 3]
```diff
@@ -235,75 +235,74 @@ func (b *Broker) loadCredentialsFrombytes(credentials []byte) error {
...
-	})
-	if creds == nil {
-		return nil, "", errors.New("unable to get EC2 role credentials")
+	client := imds.NewFromConfig(cfg)
+	getRegionInput := imds.GetRegionInput{}
+	regionOutput, err := client.GetRegion(ctx, &getRegionInput)
+	if err != nil {
+		return provider, defaultRegion, err
 	}
-	b.logger.Println("Created credentials object from metadata")
```
Comment (rgooch) [inline broker/aws/impl.go:268]: Why remove this logging? It's useful.

File: broker/aws/impl.go [hunk 4]
```diff
@@ -345,19 +349,25 @@ func (b *Broker) withSessionGetAWSRoleList(validSession *session.Session, accoun
 	}
 	defer b.listRolesSemaphore.Release(1)
 
+	b.logger.Debugf(1, "withAWSCredentialsProviderGetAWSRoleList: after semapore adquired")
```
Comment (rgooch) [inline broker/aws/impl.go:352]: s/adquired/acquired/

File: broker/aws/impl.go [hunk 5]
```diff
@@ -331,9 +330,14 @@ func (b *Broker) withProfileAssumeRole(accountName string, profileName string,
 
 const getAWSRolesTimeout = 10 * time.Second
 
-func (b *Broker) withSessionGetAWSRoleList(validSession *session.Session, accountName string) ([]string, error) {
-	iamClient := iam.New(validSession)
-	var maxItems int64
+func (b *Broker) withAWSCredentialsProviderGetAWSRoleList(credentialsProvider aws.CredentialsProvider, awsRegion string, accountName string) ([]string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithCredentialsProvider(credentialsProvider), config.WithRegion(awsRegion))
+	if err != nil {
+		b.logger.Debugf(1, "withAWSCredentialsProviderGetAWSRoleList: fail to create new config err=%s", err)
```
Comment (rgooch) [inline broker/aws/impl.go:336]: s/fail/failed/

File: broker/aws/impl.go [hunk 6]
```diff
@@ -345,19 +349,25 @@ func (b *Broker) withSessionGetAWSRoleList(validSession *session.Session, accoun
...
-					roleNames = append(roleNames, *role.RoleName)
-				}
-				return !lastPage
-			})
-		c <- err
+		paginator := iam.NewListRolesPaginator(iamClient, &listRolesInput)
+		for paginator.HasMorePages() {
+			listRolesOutput, err := paginator.NextPage(ctx)
+			if err != nil {
+				b.logger.Debugf(1, "withAWSCredentialsProviderGetAWSRoleList: faled to get roles, account=%s err=%s", accountName, err)
```
Comment (rgooch) [inline broker/aws/impl.go:362]: s/faled/failed/

**Review by rgooch — CHANGES_REQUESTED** [review]

**Review by rgooch — APPROVED** [review]

**Pull request merged**